db/db_impl_write.cc

```diff
@@ -655,6 +782,10 @@ Status DBImpl::SwitchMemtable(ColumnFamilyData* cfd, WriteContext* context) {
   log::Writer* new_log = nullptr;
   MemTable* new_mem = nullptr;
 
+  // In case of pipelined write is enabled, wait for all pending memtable
+  // writers.
+  write_thread_.WaitForMemTableWriters();
```

Comment:
This is a nice approach!

Other calling of the function is guarded by enable_pipelined_write_. Maybe we should do the same here to be consistent.